src/SharedComponents/AstroVirtualAssistant/AstroVirtualAssistant.tsx

```diff
@@ -102,11 +104,50 @@ export const AstroVirtualAssistantLegacy: FunctionComponent<AstroVirtualAssistan
   );
 };
 
+const AsyncManager = ({
+  id,
+  manager,
+  setAsyncManagers,
+}: {
+  id: string;
+  manager: AsyncStateManager<IAIClient<Record<string, unknown>>>;
+  setAsyncManagers: React.Dispatch<React.SetStateAction<AsyncManagersMap>>;
+}) => {
+  const stateManager = manager.useStateManager();
+  const authStatus = manager.useIsAuthenticated();
+  React.useEffect(() => {
+    setAsyncManagers((v) => {
+      return {
+        ...v,
+        [id]: {
+          stateManager,
+          authStatus,
+        },
+      };
+    });
+  }, [stateManager, authStatus]);
+  return null;
```

Comment:
Yeah, this is the reason why I did not originally use hooks in the async manager. I don't really have an issue with this, however we should explore some nice way to actually be able to use "async hooks".

Reply:
FYI This is well tried pattern of using the federated hooks from OCP Console